src/apps/common/submodules/portWizard/portWizard.js

```diff
@@ -2606,6 +3063,85 @@ define(function(require) {
 			});
 		},
 
+		/**
+		 * Gets the carrier data for a list of phone numbers
+		 * @param  {Object} args
+		 * @param  {Array} args.formattedNumbers  Formatted data for the phone numbers to query
+		 * @param  {('local'|'tollFree')} args.numbersType  Type of numbers
+		 * @param  {Function} args.callback  Async.js callback
+		 */
+		portWizardLoadNumbersCarrierDataAndRequirements: function(args) {
+			var self = this,
+				formattedNumbers = args.formattedNumbers,
+				numbersType = args.numbersType,
+				callback = args.callback;
+
+			monster.waterfall([
+				function(waterfallCallback) {
+					self.portWizardCarrierSelectionGetNumberCarriers({
+						formattedNumbers: formattedNumbers,
+						success: function(carrierData) {
+							waterfallCallback(null, carrierData);
+						},
+						error: function(errorData) {
+							waterfallCallback(errorData);
+						}
+					});
+				},
+				function(numbersCarrierData, waterfallCallback) {
+					var numbersByLosingCarrier = numbersCarrierData.numbersByLosingCarrier,
+						winningCarriers = numbersCarrierData.winningCarriers,
+						losingCarriersCount = _.size(numbersByLosingCarrier),
+						isSingleLosingCarrier = losingCarriersCount === 1,
+						isSingleLosingCarrierUnknown = isSingleLosingCarrier && numbersByLosingCarrier[0].carrier === 'Unknown',
+						noWinningCarriers = isSingleLosingCarrier && _.isEmpty(winningCarriers),
+						numbersCountryCode = isSingleLosingCarrier && _.get(formattedNumbers, [0, 'country', 'code']),
+						isSameCountry = isSingleLosingCarrier && _.every(formattedNumbers, [ 'country.code', numbersCountryCode ]),
+						isCountrySupported = _.has(self.appFlags.portWizard.requirementsByCountries, numbersCountryCode),
+						carrierWarningType = !isSingleLosingCarrier
+							? 'multipleLosingCarriers'
+							: isSingleLosingCarrierUnknown
+								? 'unknownLosingCarriers'
+								: noWinningCarriers
+									? 'noWinningCarriers'
+									: !isSameCountry
+										? 'multipleCountries'
+										: isCountrySupported
+											? 'none'
+											: 'countryNotSupported';
```

Comment:
While the logic is sound, nested ternaries should be avoided with this structure (mostly because the increasing indentation).

One possible structure for nested ternaries would be:

```js
var result = condition1 ? result1
    : condition2 ? result2
    : condition3 ? result3
    : result4;
```

Note that our linter does not currently support this structure so it would need to be updated to support regular ternaries and nested ones.